api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -172,14 +172,12 @@ public void setLocation(Location location) {
 		
 		if (obsParent.hasGroupMembers()) {
 			for (Obs child : obsParent.getGroupMembers()) {
-				if (!child.getVoided()) {
-					if (!child.isObsGrouping()) {
+				if (!child.getVoided() && !child.isObsGrouping()) {
```

Comment:
I now find this harder to read than before.

Reply:
But this causes it to be nested only thrice

Reply:
But do you agree that the code is now harder to understand?

Reply:
Ya I do :pensive: how can I change it then without nesting more than 3 times

Reply:
I have no idea. But you could look for another place to apply this rule without making the code harder.